api/config/v1alpha1/envoygateway_types.go

```diff
@@ -139,11 +139,28 @@ const (
 	RedisBackendType RateLimitDatabaseBackendType = "Redis"
 )
 
-// RateLimitRedisSettings defines the configuration for connecting to
-// a Redis database.
+// RedisTLSSettings defines the TLS configuration for connecting to redis database.
+type RedisTLSSettings struct {
+	// Auth defines auth configuration to enable password-only authentication to the redis host.
+	//
+	// +optional
+	Auth string `json:"auth,omitempty"`
```

Comment:
this feels like something that should be part of a secret and not directly in the API
then the question is, should be be part a different `Opaque` secret from the `TLS` secrets for the certs or the same one ?

Reply:
or should we rm the `auth` field for now and ask users to use
```
kind: EnvoyGateway
    provider:
      type: Kubernetes
      kubernetes:
        rateLimitDeployment:
            env:
            - name: REDIS_AUTH
               valueFrom:
                 secretKeyRef:
                   name: redis-passwordd
                   key: password
```

Reply:
I will rm `auth` and `RedisTLSSettings`, using `TLSCertificateRef` in `RateLimitRedisSettings` directly


```suggestion
	// RateLimitRedisSettings defines the configuration for connecting to redis database.
type RateLimitRedisSettings struct {
	// URL of the Redis Database.
	URL string `json:"url"`

	// TLSCertificateRef defines the client certificate reference kubernetes secret for TLS connections.
	//
	// +optional
	TLSCertificateRef string `json:"tlsCertificateRef,omitempty"`
}
```